proto/config/config.proto

```diff
@@ -0,0 +1,89 @@
+syntax = "proto3";
+
+package linkerd2.config;
+
+option go_package = "github.com/linkerd/linkerd2/controller/gen/config";
+
+// ConfigStore provides an abstraction over the underlying configuration storge.
+service ConfigStore {
+  // Return the JSON-serialized global config.
+  rpc GetGlobalConfig(GetParam) returns (GlobalConfig) {}
+
+  // Return the JSON-serialized proxy config.
+  rpc GetProxyConfig(GetParam) returns (ProxyConfig) {}
+
+  // Set the global config. It overwrites the existing value.
+  rpc SetGlobalConfig(GlobalConfig) returns (Result) {}
+
+  // Set the proxy config. It overwrites the existing value.
+  rpc SetProxyConfig(ProxyConfig) returns (Result) {}
+}
+
+message GlobalConfig {
+  string linkerd_namespace = 1;
+  bool cni_enabled = 2;
+  bool tls_enabled = 3;
+}
+
+message ProxyConfig {
+  Image proxy_image = 1;
+  Image proxy_init_image = 2;
+
+  Port api_port = 3;
+  Port control_port = 4;
+  repeated Port ignore_inbound_ports = 5;
+  repeated Port ignore_outbound_ports = 6;
+  Port inbound_port = 7;
+  Port metrics_port = 8;
+  Port outbound_port = 9;
+
+  ResourceRequirements resource = 10;
+
+  TLS tls_enabled = 11;
```

Comment:
In the current auto-inject implementation, whether or not injected proxies use TLS is only configurable at control plane install time, and it applies globally to all injected deployments. In that case, it seems like we can use the TLS setting from `GlobalConfig` to satisfy the current behavior, without also need a TLS setting in this struct. It's entirely possible I'm forgetting something about the current implementation though.